test/src/test/java/hudson/model/utils/ResultWriterPublisher.java

```diff
@@ -0,0 +1,47 @@
+package hudson.model.utils;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Publisher;
+import hudson.tasks.Recorder;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+/**
+ * Wrote build status to file
+ * @author Kanstantsin Shautsou
+ */
+public class ResultWriterPublisher extends Recorder {
+    private final String fileName;
+
+    public ResultWriterPublisher(String fileName) {
+        this.fileName = fileName;
+    }
+
+    @Override
+    public BuildStepMonitor getRequiredMonitorService() {
+        return BuildStepMonitor.NONE;
+    }
+
+    @Override
+    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+        FilePath file = build.getWorkspace().child(fileName);
+        file.write(build.getResult().toString(), Charset.defaultCharset().name());
```

Comment:
This is fine, though it would be slightly more concise to just print the result in a specially-formatted message to the `listener`. Then you can use `JenkinsRule.assertBuildLogContains`.